### PR TITLE
Adds tracing

### PR DESCRIPTION
**Pull request by christopherjturner**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -19,7 +19,8 @@ export default {
     '<rootDir>/.server',
     'index.js'
   ],
-  coverageDirectory: '<rootDir>/coverage'
+  coverageDirectory: '<rootDir>/coverage',
+  transformIgnorePatterns: ['/node_modules(?!/(@defra/hapi-tracing))']
```
Comment (feedmypixel) [inline jest.config.js:23]: comment maybe around why?

`// ESM only module`

File: src/api/common/helpers/logging/trace-id-mixin.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { getTraceId } from '@defra/hapi-tracing'
```
Comment (feedmypixel) [inline src/api/common/helpers/logging/trace-id-mixin.js:1]: Not convinced on this abstraction. What if you want to add extra things in to the mixin? Considering the mixin pattern is a pino pattern IMO feels like this should just sit where it was in the loggerOptions. So its simple to see and override

**Review by feedmypixel — COMMENTED** [review]
Is it worth running this through SonarCloud?

File: src/api/common/helpers/logging/trace-id-mixin.js (hunk 2 above)
Comment (christopherjturner) [reply src/api/common/helpers/logging/trace-id-mixin.js:1]: I guess we could in-line it, fewer files etc

**Review by feedmypixel — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**